fcomm/src/lib.rs

```diff
@@ -1085,6 +1086,48 @@ mod test {
     use lurk::public_parameters::public_params;
     use lurk::z_data::{from_z_data, to_z_data};
 
+    // You have broken a snapshot test. Unlike round-trip tests, those tests check the actual format of serialized Lurk expressions,
+    // and since you broke one, it's probable that you have changed that format, which will break at least the fcomm examples.
+    // Please read the documentation on snapshot tests `https://insta.rs/docs/quickstart/`, fix the snapshot **AND**
+    // make sure `cargo nextest run --run-ignored all -E 'test(test_make_fcomm_examples)'` passes
+    #[test]
+    fn test_snapshot_serialized_expressions() {
+        let function_source: &str = "(letrec ((secret 12345) (a (lambda (acc x) (let ((acc (+ acc x))) (cons acc (hide secret (a acc))))))) (a 0))";
+        let function_inputs: &str = "(+ 1 2)";
+        let committed_expression = CommittedExpression::<S1> {
+            expr: LurkPtr::Source(function_source.into()),
+            secret: None,
+            commitment: None,
+        };
+        assert_json_snapshot!(committed_expression);
+
+        let input = Expression::<S1> {
+            expr: LurkPtr::Source(function_inputs.into()),
+        };
+        assert_json_snapshot!(input);
+
+        let c = Commitment {
+            comm: S1::from(123),
+        };
+        assert_json_snapshot!(c);
+
+        let req = OpeningRequest {
+            input,
+            commitment: c.clone(),
+            chain: true,
+        };
+        assert_json_snapshot!(req);
+
+        let opening = Opening {
+            input: function_inputs.to_owned(),
+            output: function_inputs.to_owned(),
+            status: Status::Error,
+            commitment: c,
+            new_commitment: None,
+        };
```

Comment:
If we use the correct input, then we can hope for an opening that yields an interesting result, rather than an error.

Reply:
Yes. I consider this to be out of scope.